### PR TITLE
Add pgp key pair secret type

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade/issues/11711

### CODE REVIEW

File: src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            _console = console;
+        }
+
+        protected override async Task<SecretData> RotateValue(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            // Adding this to suppress the warning about not having the 'await' operator in an async method
+            await context.GetSecretValue(new SecretReference(context.SecretName));
+
+            throw new HumanInterventionRequiredException($"Pgp key pair secret rotation required. Human intervention required. Please go to https://aka.ms/signalr-java-publishing and follow the instructions at the bottom of the page");
```
Comment (riarenas) [inline src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: This page is heavily skewed towards the signalR pgp key. If you're going to provide instructions in a wiki, it should be to a page for generic pgp key generation.

If this particular secret can't be cycled and instead needs to be extended forever, that's a note for the secret itself, not for secret-manager.

Comment (riarenas) [inline src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: Shouldn't there be more code here? I would expect that after the help message is thrown, a prompt asking for the new values will show up. You can check an example here: https://github.com/dotnet/arcade-services/blob/main/src/Microsoft.DncEng.SecretManager/SecretTypes/TextSecretType.cs

File: src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    
+        public PgpKeyPair(IConsole console)
+        {
+            _console = console;
+        }
+
+        protected override async Task<SecretData> RotateValue(Parameters parameters, RotationContext context, CancellationToken cancellationToken)
+        {
+            // Adding this to suppress the warning about not having the 'await' operator in an async method
+            await context.GetSecretValue(new SecretReference(context.SecretName));
```
Comment (alexperovich) [inline src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:26]: Please remove this and just suppress the warning with pragma.
Or even better just return Task.FromResult()

File: src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: This is not how this is supposed to work. This exception should be thrown if the console isn't interactive.

If the console is interactive this needs to prompt for new values.

**Review by alexperovich — CHANGES_REQUESTED** [review]
Please make this actually prompt for a new value when the user is using the program.

File: src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs (hunk 1 above)
Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: We talked to Barry about this, extending the key also changes it's private and public key, so it is cycling it, without having to create a new one and doing so isn't dangerous, as long as the key hasn't been compromised. It is something we can consider doing for all future PGP keys

Comment (riarenas) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: If this is the approach for all future secrets of this type, then I'd like to see the code to automatically rotate included in this PR. Let's not add new secret types that are prone to manual mistakes and have so many requisites.

Comment (riarenas) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: At the very least, this still needs to point to a generic document that is not related to signalR.

Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: Before we commit to adding PGP key rotation to the Secret Manager, I'd like to try to the other, ESRP route.
I will create a separate page for rotating PGP keys in general

Comment (riarenas) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: That sounds completely reasonable.

Comment (riarenas) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: Would it make sense to hold on this PR until we're sure we need to add this type of secret to our vaults? You mentioned you would be looking into the ESRP task early next year after the release?

Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/SecretTypes/PgpKeyPair.cs:28]: I think it would, let's go with that

**Comment by dkurepa** [comment]
this shouldn't be needed. We will have the PGP created and maintained by the ESRP team

**Pull request closed without merging**